File: Project/CocoaLUTTests/ParserTests.m
//
//  ParserTests.m
//  CocoaLUT
//
//  Created by Wil Gieseler on 6/25/14.
//  Copyright (c) 2014 Wil Gieseler. All rights reserved.
//

#import <XCTest/XCTest.h>
#import <CocoaLUT/CocoaLUT.h>

@interface ParserTests : XCTestCase

@end

@implementation ParserTests

- (LUT *)loadLUT:(NSString *)name extension:(NSString *)ext {
    return [LUT LUTFromURL:[[NSBundle bundleForClass: [self class]] URLForResource:name withExtension:ext]];
}

- (void)testParseCubeCrosstalk {
    XCTAssertEqual([self loadLUT:@"crosstalk" extension:@"cube"].size, 17);
}

- (void)testParseCubeHalfredIridas {
    XCTAssertEqual([self loadLUT:@"halfred_iridas" extension:@"cube"].size, 17);
}

//- (void)testParseCubeIridas {
//    XCTAssertEqual([self loadLUT:@"iridas" extension:@"cube"].size, 17);
//}

- (void)testParse3DLCrosstalk {
    XCTAssertEqual([self loadLUT:@"crosstalk" extension:@"3dl"].size, 17);
}

- (void)testParse3DLHalfredTruelight {
    XCTAssertEqual([self loadLUT:@"halfred_truelight" extension:@"3dl"].size, 17);
}

- (void)testParse3DLHalfredTruelightLog {
    XCTAssertEqual([self loadLUT:@"halfred_truelight_log" extension:@"3dl"].size, 17);
}

@end

File: Project/CocoaLUTTests/ParserTests.swift
//
//  CocoaLUTTests.swift
//  CocoaLUTTests
//
//  Created by Wil Gieseler on 6/25/14.
//  Copyright (c) 2014 Wil Gieseler. All rights reserved.
//

import XCTest
import Cocoa
import CocoaLUT

class ParserTests: XCTestCase {
    
    func LUTFromBundle(name: String, ext: String) -> LUT {
        let url = NSBundle(forClass: self.dynamicType).URLForResource(name, withExtension: ext)
        return LUT(fromURL: url)
    }
    
    func testParseCubeCrosstalk() {
        XCTAssertEqual(LUTFromBundle("crosstalk", ext: "cube").size, 17)
    }
    
    func testParseCubeHalfredIridas() {
        XCTAssertEqual(LUTFromBundle("halfred_iridas", ext: "cube").size, 17)
    }
    
    func testParseCubeIridas() {
        XCTAssertEqual(LUTFromBundle("iridas", ext: "").size, 2)
    }
    
    func testParse3DLCrosstalk() {
        XCTAssertEqual(LUTFromBundle("crosstalk", ext: "3dl").size, 17)
    }
    
    func testParse3DLHalfredTruelight() {
        XCTAssertEqual(LUTFromBundle("halfred_truelight", ext: "3dl").size, 17)
    }
    
    func testParse3DLHalfredTruelightLog() {
        XCTAssertEqual(LUTFromBundle("halfred_truelight_log", ext: "3dl").size, 17)
    }

}
